auth0/resource_auth0_log_stream.go

```diff
@@ -0,0 +1,366 @@
+	}
+
+	d.SetId(ls.GetID())
+	d.Set("name", ls.Name)
+	d.Set("status", ls.Status)
+	d.Set("type", ls.Type)
+	d.Set("sink", flattenLogStreamSink(d, ls.Sink))
+	return nil
+}
+
+func updateLogStream(d *schema.ResourceData, m interface{}) error {
+	ls := expandLogStream(d)
+	fmt.Printf("UPDATE: ---------------> %+v\n", ls)
+	api := m.(*management.Management)
+	err := api.LogStream.Update(d.Id(), ls)
+	if err != nil {
+		return err
+	}
+	return readLogStream(d, m)
+}
+
+func deleteLogStream(d *schema.ResourceData, m interface{}) error {
+	api := m.(*management.Management)
+	err := api.LogStream.Delete(d.Id())
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok {
+			if mErr.Status() == http.StatusNotFound {
+				d.SetId("")
+				return nil
+			}
+		}
+	}
+	return err
+}
+
+func flattenLogStreamSink(d ResourceData, sink interface{}) []interface{} {
+
+	var m interface{}
+
+	switch o := sink.(type) {
+	case *management.LogStreamSinkAmazonEventBridge:
+		m = flattenLogStreamSinkAmazonEventBridge(o)
+	case *management.LogStreamSinkAzureEventGrid:
+		m = flattenLogStreamSinkAzureEventGrid(o)
+	case *management.LogStreamSinkHTTP:
+		m = flattenLogStreamSinkHTTP(o)
+	case *management.LogStreamSinkDatadog:
+		m = flattenLogStreamSinkDatadog(o)
+	case *management.LogStreamSinkSplunk:
+		m = flattenLogStreamSinkSplunk(o)
+	}
+	return []interface{}{m}
+}
+
+func flattenLogStreamSinkAmazonEventBridge(o *management.LogStreamSinkAmazonEventBridge) interface{} {
+	return map[string]interface{}{
+		"aws_account_id":           o.GetAccountID(),
+		"aws_region":               o.GetRegion(),
+		"aws_partner_event_source": o.GetPartnerEventSource(),
+	}
+}
+
+func flattenLogStreamSinkAzureEventGrid(o *management.LogStreamSinkAzureEventGrid) interface{} {
+	return map[string]interface{}{
+		"azure_subscription_id": o.GetSubscriptionID(),
+		"azure_resource_group":  o.GetResourceGroup(),
+		"azure_region":          o.GetRegion(),
+		"azure_partner_topic":   o.GetPartnerTopic(),
+	}
+}
+
+func flattenLogStreamSinkHTTP(o *management.LogStreamSinkHTTP) interface{} {
+	return map[string]interface{}{
+		"http_endpoint":       o.GetEndpoint(),
+		"http_contentFormat":  o.GetContentFormat(),
+		"http_contentType":    o.GetContentType(),
+		"http_authorization":  o.GetAuthorization(),
+		"http_custom_headers": o.CustomHeaders,
+	}
+}
+
+func flattenLogStreamSinkDatadog(o *management.LogStreamSinkDatadog) interface{} {
+	return map[string]interface{}{
+		"datadog_region":  o.GetRegion(),
+		"datadog_api_key": o.GetAPIKey(),
+	}
+}
+
+func flattenLogStreamSinkSplunk(o *management.LogStreamSinkSplunk) interface{} {
+	return map[string]interface{}{
+		"splunk_domain": o.GetDomain(),
+		"splunk_token":  o.GetToken(),
+		"splunk_port":   o.GetPort(),
+		"splunk_secure": o.GetSecure(),
+	}
+}
+func expandLogStream(d ResourceData) *management.LogStream {
+
+	ls := &management.LogStream{
+		Name:   String(d, "name", HasChange()),
+		Type:   String(d, "type", HasChange()),
```

Comment:
Probably better to use `IsNewResource` as `Type` isn't allowed to change. Also `Name` can be present any time.

```suggestion
		Name:   String(d, "name"),
		Type:   String(d, "type", IsNewResource()),
```